src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs

```diff
@@ -335,6 +335,7 @@ out List<KnownRuntimePack> knownRuntimePacksForTargetFramework
                 var runtimeRequiredByDeployment
                     = (SelfContained || ReadyToRunEnabled) &&
                       !string.IsNullOrEmpty(RuntimeIdentifier) &&
+                      RuntimeIdentifier != "any" &&
                       selectedRuntimePack != null &&
```

Comment:
In DependencyContextBuilder, I wonder if we could skip the check for nativeRuntimeTargetsFiles if the RID is Any?

Reply:
I didn't quite understand what you meant here, but I did see a related change that we needed to make around a check on `_isPortable` that I've updated too.